microbench/src/main/java/io/netty/microbenchmark/common/IsValidIpV6Benchmark.java

```diff
@@ -0,0 +1,371 @@
+                            return false;
+                        }
+                    }
+                }
+
+                if (!isValidIp4Word(word.toString())) {
+                    return false;
+                }
+                word.delete(0, word.length());
+                break;
+
+            case ':':
+                // FIX "IP6 mechanism syntax #ip6-bad1"
+                // An IPV6 address cannot start with a single ":".
+                // Either it can start with "::" or with a number.
+                if (i == startOffset && (ipAddress.length() <= i || ipAddress.charAt(i + 1) != ':')) {
+                    return false;
+                }
+                // END FIX "IP6 mechanism syntax #ip6-bad1"
+                numberOfColons ++;
+                if (numberOfColons > 7) {
+                    return false;
+                }
+                if (numberOfPeriods > 0) {
+                    return false;
+                }
+                if (prevChar == ':') {
+                    if (doubleColon) {
+                        return false;
+                    }
+                    doubleColon = true;
+                }
+                word.delete(0, word.length());
+                break;
+
+            default:
+                if (word != null && word.length() > 3) {
+                    return false;
+                }
+                if (!isValidHexChar(c)) {
+                    return false;
+                }
+                word.append(c);
+            }
+        }
+
+        // Check if we have an IPv4 ending
+        if (numberOfPeriods > 0) {
+            // There is a test case with 7 colons and valid ipv4 this should resolve it
+            if (numberOfPeriods != 3 || !(isValidIp4Word(word.toString()) && (numberOfColons < 7 || doubleColon))) {
+                return false;
+            }
+        } else {
+            // If we're at then end and we haven't had 7 colons then there is a
+            // problem unless we encountered a doubleColon
+            if (numberOfColons != 7 && !doubleColon) {
+                return false;
+            }
+
+            // If we have an empty word at the end, it means we ended in either
+            // a : or a .
+            // If we did not end in :: then this is invalid
+            if (word.length() == 0 && ipAddress.charAt(length - 1 - startOffset) == ':' &&
+                ipAddress.charAt(length - 2 - startOffset) != ':') {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    // New
+
+    private static boolean isValidIpV4WordNew(CharSequence word, int start, int endExclusive) {
+        int len = endExclusive - start;
+        char c0, c1, c2;
+        if (len < 1 || len > 3 || (c0 = word.charAt(start)) < '0') {
+            return false;
+        }
+        if (len == 3) {
+            return (c1 = word.charAt(start + 1)) >= '0' &&
+                   (c2 = word.charAt(start + 2)) >= '0' &&
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (9) used in the expression (maximum allowed 5). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1067)